File: datasets/Python/Deepseek/DataStructures/easy4.py
def getMax(operations):
    stack = []
    max_stack = []
    result = []
    for op in operations:
        if op.startswith('1'):
            num = int(op.split()[1])
            stack.append(num)
            if not max_stack or num >= max_stack[-1]:
                max_stack.append(num)
        elif op == '2':
            if stack.pop() == max_stack[-1]:
                max_stack.pop()
        elif op == '3':
            result.append(max_stack[-1])
    return result

n = int(input())
operations = [input().strip() for _ in range(n)]
output = getMax(operations)
for num in output:
    print(num)

File: datasets/C++/Deepseek/DataStructures/easy4.cpp
#include <iostream>
#include <vector>
#include <stack>
#include <string>
#include <sstream>
using namespace std;

vector<int> getMax(vector<string> operations) {
    stack<int> s;
    stack<int> max_s;
    vector<int> result;
    for (string op : operations) {
        if (op[0] == '1') {
            int num = stoi(op.substr(2));
            s.push(num);
            if (max_s.empty() || num >= max_s.top()) {
                max_s.push(num);
            }
        } else if (op == "2") {
            if (s.top() == max_s.top()) {
                max_s.pop();
            }
            s.pop();
        } else if (op == "3") {
            result.push_back(max_s.top());
        }
    }
    return result;
}

int main() {
    int n;
    cin >> n;
    cin.ignore();
    vector<string> operations(n);
    for (int i = 0; i < n; i++) {
        getline(cin, operations[i]);
    }
    vector<int> output = getMax(operations);
    for (int num : output) {
        cout << num << endl;
    }
    return 0;
}